content/docs/guides/adopting/import.md

```diff
@@ -18,204 +20,399 @@ When working with existing resources, there are two primary scenarios:
+
+import (
+	"github.com/pulumi/pulumi-aws/sdk/v3/go/aws/s3"
+	"github.com/pulumi/pulumi/sdk/v2/go/pulumi"
 )
-if err != nil {
-    return err
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+		_, err := s3.NewBucket(ctx, "infra-logs", &s3.BucketArgs{
+			Acl:          pulumi.String("private"),
+			Bucket:       pulumi.String("company-infra-logs"),
+			ForceDestroy: pulumi.Bool(false),
+		}, pulumi.Protect(true))
+		if err != nil {
+			return err
+		}
+		return nil
+	})
 }
 ```
 
-{{% /choosable %}}
-{{% choosable language csharp %}}
+{{< /choosable >}}
+{{< choosable language csharp >}}
 
 ```csharp
-var group = new SecurityGroup("my-sg",
-    new SecurityGroupArgs {
-        Name = "my-sg-62a569b",
-        Description = "Enable HTTP access",
-        Ingress = {
-            new SecurityGroupIngressArgs {
-                Protocol = "tcp",
-                FromPort = 80,
-                ToPort = 80,
-                CidrBlocks = { "0.0.0.0/0" }
-            }
-        }
-    },
-    new CustomResourceOptions {
-        ImportId = "sg-04aeda9a214730248"
+using Pulumi;
+using Aws = Pulumi.Aws;
+
+class MyStack : Stack
+{
+    public MyStack()
+    {
+        var demo = new Aws.S3.Bucket("infra-logs", new Aws.S3.BucketArgs
+        {
+            Acl = "private",
+            Bucket = "company-infra-logs",
+            ForceDestroy = false,
+        }, new CustomResourceOptions
+        {
+            Protect = true,
+        });
     }
-);
-```
 
-{{% /choosable %}}
+}
+```
 
+{{< /choosable >}}
 {{< /chooser >}}
 
-> **Note:** Import IDs are resource specific. The ID to use is the same as the ID that gets assigned when Pulumi has provisioned a resource of that type from scratch.
-
-When Pulumi first sees a resource with an `import` option set (in this case `my-sg`), it will adopt the existing resource by querying the target cloud provider for a resource of that type with the given ID, instead of creating a new resource as usual.
-
-To perform the import, run `pulumi up` as usual, and you will see `=` instead of the usual `+`, indicating an import operation:
+> Your Pulumi stack must be configured correctly---e.g., in this case, the correct AWS region---otherwise the resource will not be found.
 
+After successfully importing a resource, you can run `pulumi up`, and all subsequent operations will behave as though Pulumi provisioned
+the resource from the outset. When Pulumi performs an import, the resource is added to the Pulumi [state](https://www.pulumi.com/docs/intro/concepts/state/#state),
+it is marked as a [protected](https://www.pulumi.com/docs/intro/concepts/programming-model/#protect) resource (by default),
+and it will emit the generated code that the user can add to their program before running a `pulumi up`. The resources are
+marked as protected to ensure that imported infrastructure is not accidentally deleted if the user forgets to include
+the code for the resource in their program before doing a deployment.
```

Comment:
Some of the added text here was covered above, so I took a stab at summarizing existing information and keeping the new info in about the protected nature of these resources.

```suggestion
After successfully importing a resource and adding the generated code to your program, you can run `pulumi up` and all subsequent operations will behave as though Pulumi provisioned
the resource from the outset. The resource is added to the Pulumi [state](https://www.pulumi.com/docs/intro/concepts/state/#state),
and marked as a [protected](https://www.pulumi.com/docs/intro/concepts/programming-model/#protect) resource (by default) to ensure that imported infrastructure is not accidentally deleted if the user forgets to include the code for the resource in their program before doing a deployment.
```